src/excmds.ts

```diff
@@ -3072,7 +3072,38 @@ export async function bmark(url?: string, ...titlearr: string[]) {
     dupbmarks.map(bookmark => browser.bookmarks.remove(bookmark.id))
     if (dupbmarks.length != 0) return
     const path = title.substring(0, title.lastIndexOf("/") + 1)
-    // TODO: if title is blank, get it from the page.
+
+
+    //this whole block grabs the current tab's title
+    //based on an async call since document.title won't have any meaning here
+    //
+    var currTitle;
+    function getCurrTabTitle(tabs) {
+        //function callback for promise for getting title of current tab
+        //requires [tabs] permission which tridactyl should have grabbed anyway
+        if (tabs[0]) {
+            var currentTab = tabs[0];
+            currTitle = currentTab.title
+        }
+    }
+    var titleSourceTab = browser.tabs.query({active: true, currentWindow: true});
+    //TODO: define and implement sane behaviour for if the title is blank, but the URL
+    //      that is being bookmarked is not actually the current tab. just using 
+    //      await activeTab().title doesn't make sense for that, we'll need to change the tab query
+    //      immediately above to search for the url.  I guess then just go with arr[0] for the winner
+    //      The possibility also exists that someone tries to bmark a URL thats not a current tab anywhere
+    //      so sane behaviour needs to be defined for that too.  Load the page headless and extract URL?
+    //      Make the title the URL?
+    titleSourceTab.then(getCurrTabTitle);
+    await titleSourceTab;
+
+    // if title is blank, get it from the page.
+    if (title == "") {
+        title = currTitle
+    }
+
+
+
     if (path != "") {
```

Comment:
We probably need to skip this step if we're getting the title from the page - we don't want "BBC News - '/' becomes part of the alphabet" to end up in a "BBC News - " folder.

Reply:
`path` is actually determined before the function ever hits the default title code, so there's no issue here.  Cases are:
* Bookmark function is passed a 'title' including a path, in which case the first few lines of code pick it up and the default blank title code never triggers, then the path processing code triggers
* Bookmark function is passed a title with no path.  Default title code never triggers, and neither does the path processing code.  
* Bookmark function is passed no title at all.  The default title code triggers and grabs the current tab's title.  Whatever that title is doesn't matter, as the path var is already just nothing and thus the path processing block doesn't trigger.